### PR TITLE
[OM] Add path append op

### PR DESCRIPTION
**Pull request by prithayan**
Add the path append op to OM dialect.

### CODE REVIEW

File: test/Dialect/OM/round-trip.mlir [hunk 1]
```diff
@@ -252,9 +252,16 @@ hw.module @PathModule() {
...
-om.class @Path() {
+om.class @Path(%p: !om.path, %p2: !om.path) {
   // CHECK: %0 = om.path reference @HierPath
   %0 = om.path reference @HierPath
+  // CHECK:  om.path_append %p, %0
+  %1 = om.path_append %p, %0
+  %2 = om.path_append %p, %p2
+  om.class.field @path1, %2 : !om.path
+  //CHECK:  %2 = om.path_append %p, %p2
+  //CHECK:  om.class.field @path1, %2 : !om.path
```
Comment (uenoku) [inline test/Dialect/OM/round-trip.mlir:263]: nit:
```suggestion
  // CHECK:  %2 = om.path_append %p, %p2
  // CHECK:  om.class.field @path1, %2 : !om.path
```

Also please use substitutions for `%0` and `%1`.

**Review by uenoku — APPROVED** [review]

File: include/circt/Dialect/OM/OMOps.td [hunk 2]
```diff
@@ -338,4 +338,21 @@ def AnyCastOp : OMOp<"any_cast", [Pure]> {
...
+  let description = [{
+    Appends the path to the root path. The root path must not have a component.
+    The appended path must begin where the root path ends.
+
+    Example:
+    ` ` `mlir
+    %root = om.constant #om.path<"Foo/bar:Bar">
+    %path = om.constant #om.path<"Bar/baz:Baz>output">
+    %0 = om.path_append %root, %path
+    ` ` `
```
Comment (youngar) [inline include/circt/Dialect/OM/OMOps.td:352]: spaces between '`' that shouldn't be there.

File: test/Dialect/OM/round-trip.mlir [hunk 3]
```diff
@@ -252,9 +252,15 @@ hw.module @PathModule() {
...
-  // CHECK: %0 = om.path reference @HierPath
+om.class @Path(%p: !om.path, %p2: !om.path) {
+  // CHECK: %[[v0:.+]] = om.path reference @HierPath
   %0 = om.path reference @HierPath
+  // CHECK:  om.path_append %p, %[[v0]]
+  %1 = om.path_append %p, %0
+  %2 = om.path_append %p, %p2
+  om.class.field @path1, %2 : !om.path
+  // CHECK:  %[[v2:.+]] = om.path_append %p, %p2
+  // CHECK:  om.class.field @path1, %[[v2]] : !om.path
```
Comment (youngar) [inline test/Dialect/OM/round-trip.mlir:263]: some of the CHECK are leading, other are trailing. Seems like the above tests are in the leading style. Also seems like we don't need everything here, so I suggest the following:
```suggestion
om.class @Path(%p: !om.path) {
  // CHECK: %[[v0:.+]] = om.path reference @HierPath
  %0 = om.path reference @HierPath
  // CHECK:  %[[v1:.+]] = om.path_append %p, %[[v0]]
  %1 = om.path_append %p, %0
  // CHECK:  om.class.field @path1, %[[v1]] : !om.path
  om.class.field @path1, %1 : !om.path
```

**Review by youngar — APPROVED** [review]
LGTM thanks for adding this

**Comment by prithayan** [comment]
Thanks @youngar, @uenoku for the review. Addressed the comments.

**Pull request merged**